lib/dictBuilder/cover.c

```diff
@@ -958,6 +994,10 @@ ZDICTLIB_API size_t ZDICT_optimizeTrainFromBuffer_cover(
   POOL_ctx *pool = NULL;
 
   /* Checks */
+  if (splitPoint <= 0 || splitPoint > 1) {
```

Comment:
Lets only make `0.0` the default. Otherwise this check will never fail, and people might get unexpected results if they pass 99 for example.

Reply:
When checking if splitPoint is 0.0, is it ok that I use <=0.0 instead since the number from cli can't be negative? Or should I do the floating point comparison with epsilon to check if this is 0.0?

Reply:
You should use `== 0.0`, the API doesn't have the same restriction. We can use equal here, since we know we will be passing exactly `0.0`.